### ISSUE TITLE
how to configure balance to create multiple connections for a single address

### ISSUE THREAD

**Issue by jiacai2050** [grpc/grpc-go#3005]
In [this article](https://grpc.io/blog/grpc_on_http2/), it says:

> Alternatively, a user who wants multiple connections but knows that the host name will only resolve to one address might have their balancer create connections against each address 10 times to ensure that multiple connections are used.

But I have no idea how to do this using [balancer api](https://godoc.org/google.golang.org/grpc/balancer)

**Comment by menghanl** [comment]
You can call `ClientConn.NewSubConn` multiple times with the same address.

**Comment by jiacai2050** [comment]
Hi @menghanl thanks for you quick reply. However I still have trouble implementing this. 

First of all, You mean [balancer#ClientConn](https://godoc.org/google.golang.org/grpc/balancer#ClientConn), not [grpc#ClientConn](https://godoc.org/google.golang.org/grpc#ClientConn), How can I  call `ClientConn.NewSubConn`, Do I need implement a custom balancer? if so, how to do this?
I can only find [example about custom Resolver](https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/client/main.go).
 After go through #1694, It seems that I need to call [balancer#Register](https://godoc.org/google.golang.org/grpc/balancer#Register).
Could you give an example?

**Comment by menghanl** [comment]
Yes, you will need to implement a custom balancer. Take a look at [roundrobin](https://github.com/grpc/grpc-go/blob/master/balancer/roundrobin/roundrobin.go) and [balancer base](https://github.com/grpc/grpc-go/blob/master/balancer/base/balancer.go) as examples.

**Comment by jiacai2050** [comment]
> do you have any specific reasons to create multiple connections to the same address?

AKAIK, multiple connections are required to take advantage of multi-core processors.

**Comment by dfawley** [comment]
That is false.  You can create and use multiple streams over a single connection with HTTP/2 (which gRPC is built on), and one connection should be capable of saturating the network.  There are only a few, limited situations where multiple connections to one address is helpful.  E.g.: the address is a reverse proxy that fans out to multiple backends for load balancing purposes.

**Comment by jiacai2050** [comment]
@dfawley I agree with you one http2 long-live connection can fully saturate the bandwidth, what I mean is to balance pressure of tcp packet parsing on multiple cpus, which is like your proxy example.
